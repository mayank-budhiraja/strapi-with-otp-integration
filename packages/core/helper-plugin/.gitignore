# Don't check auto-generated stuff into git
dist
coverage
build
node_modules
stats.json
package-lock.json
yarn-error.log


# Cruft
.DS_Store
npm-debug.log
.idea
stats.html