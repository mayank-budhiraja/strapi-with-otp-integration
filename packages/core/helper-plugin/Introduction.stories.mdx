import { Meta } from '@storybook/addon-docs';

<Meta title="Introduction" />

# Welcome to the Helper plugin documentation

Here you will find the documentation related to the complex components that are used in the Strapi admin panel.
