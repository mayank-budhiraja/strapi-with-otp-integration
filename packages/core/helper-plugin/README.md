# Strapi Helper Plugin

## Description

Helper to develop Strapi plugins.
