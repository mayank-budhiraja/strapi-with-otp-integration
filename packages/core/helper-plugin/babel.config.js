const defaultPresets = [
  [
    '@babel/preset-env',
    {
      modules: 'commonjs',
      loose: true,
    },
  ],
];

module.exports = {
  presets: defaultPresets.concat(['@babel/preset-react', '@babel/preset-flow']),
  plugins: [
    ['@babel/plugin-proposal-class-properties', { loose: true }],
    ['@babel/plugin-proposal-private-property-in-object', { loose: true }],
    ['@babel/plugin-proposal-private-methods', { loose: true }],
    '@babel/plugin-proposal-export-default-from',
    '@babel/plugin-proposal-export-namespace-from',
    '@babel/plugin-proposal-function-bind',
  ],
};
