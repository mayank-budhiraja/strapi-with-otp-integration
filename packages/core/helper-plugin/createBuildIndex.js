const fs = require('fs-extra');
const path = require('path');

const content = `
'use strict';

if (process.env.NODE_ENV === "production") {
  module.exports = require("./helper-plugin.production.js");
} else {
  module.exports = require("./helper-plugin.development.js");
}
`;

const createIndexFile = async dest => {
  await fs.ensureDir(path.resolve(dest, 'build'));
  fs.writeFile(path.resolve(dest, 'build', 'index.js'), content);
};

createIndexFile(__dirname);
