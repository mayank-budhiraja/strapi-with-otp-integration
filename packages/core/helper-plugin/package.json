{
  "name": "@strapi/helper-plugin",
  "version": "4.1.12",
  "description": "Helper for Strapi plugins development",
  "repository": {
    "type": "git",
    "url": "https://github.com/strapi/strapi.git",
    "directory": "packages/core/helper-plugin"
  },
  "license": "SEE LICENSE IN LICENSE",
  "author": {
    "name": "Strapi Solutions SAS",
    "email": "hi@strapi.io",
    "url": "https://strapi.io"
  },
  "maintainers": [
    {
      "name": "Strapi Solutions SAS",
      "email": "hi@strapi.io",
      "url": "https://strapi.io"
    }
  ],
  "main": "build/index.js",
  "module": "lib/src/index.js",
  "files": [
    "build"
  ],
  "scripts": {
    "prebuild": "rimraf build",
    "build": "yarn build:development && yarn build:production && yarn create:index",
    "build-storybook": "build-storybook",
    "build:development": "cross-env NODE_ENV=development webpack-cli",
    "build:production": "cross-env NODE_ENV=production webpack-cli",
    "create:index": "node ./createBuildIndex.js",
    "prepublishOnly": "npm run build",
    "storybook": "start-storybook -p 6006",
    "test:front": "jest --config ./jest.config.front.js",
    "test:front:watch": "jest --config ./jest.config.front.js --watchAll",
    "watch": "yarn create:index && cross-env NODE_ENV=development webpack-cli -w"
  },
  "dependencies": {
    "@fortawesome/fontawesome-free": "^5.15.2",
    "@fortawesome/fontawesome-svg-core": "^1.2.35",
    "@fortawesome/free-brands-svg-icons": "^5.15.2",
    "@fortawesome/free-solid-svg-icons": "^5.15.3",
    "@fortawesome/react-fontawesome": "^0.1.14",
    "axios": "0.25.0",
    "babel-plugin-styled-components": "2.0.2",
    "formik": "2.2.9",
    "invariant": "^2.2.1",
    "lodash": "4.17.21",
    "match-sorter": "^4.0.2",
    "mini-css-extract-plugin": "2.4.4",
    "moment": "^2.29.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-helmet": "^6.1.0",
    "react-intl": "5.20.2",
    "react-router": "^5.2.0",
    "react-router-dom": "5.2.0",
    "styled-components": "5.3.3",
    "whatwg-fetch": "^3.6.2"
  },
  "devDependencies": {
    "@babel/core": "7.16.7",
    "@babel/plugin-proposal-class-properties": "7.16.7",
    "@babel/plugin-proposal-export-default-from": "7.16.7",
    "@babel/plugin-proposal-function-bind": "7.16.7",
    "@babel/plugin-transform-runtime": "7.16.7",
    "@babel/preset-env": "7.16.7",
    "@babel/preset-react": "7.16.7",
    "@babel/runtime": "7.16.7",
    "@storybook/addon-actions": "6.4.10",
    "@storybook/addon-essentials": "6.4.10",
    "@storybook/addon-links": "6.4.10",
    "@storybook/builder-webpack5": "6.4.10",
    "@storybook/manager-webpack5": "6.4.10",
    "@storybook/react": "^6.3.7",
    "@strapi/design-system": "1.1.1",
    "@strapi/icons": "1.1.1",
    "babel-loader": "^8.2.2",
    "cross-env": "^7.0.3",
    "date-fns": "2.22.1",
    "rimraf": "3.0.2"
  },
  "peerDependencies": {
    "formik": "^2.2.6",
    "immer": "9.0.6",
    "qs": "6.10.1",
    "react-select": "^4.0.2"
  },
  "engines": {
    "node": ">=12.22.0 <=16.x.x",
    "npm": ">=6.0.0"
  }
}
