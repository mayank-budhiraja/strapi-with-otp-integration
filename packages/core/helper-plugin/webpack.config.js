const webpack = require('webpack');
const path = require('path');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');
const packageJson = require('./package.json');

const nodeModules = [];
[
  ...Object.keys(packageJson.dependencies),
  ...Object.keys(packageJson.peerDependencies),
  ...Object.keys(packageJson.devDependencies),
].forEach(module => {
  nodeModules.push(new RegExp(`^${module}(/.+)?$`));
});

module.exports = {
  entry: `${__dirname}/lib/src/index.js`,
  externals: nodeModules,
  mode: process.env.NODE_ENV,
  devtool: process.env.NODE_ENV === 'development' ? 'eval-source-map' : false,
  output: {
    path: `${__dirname}/build`,
    filename: `helper-plugin.${process.env.NODE_ENV}.js`,
    library: {
      name: 'helperPlugin',
      type: 'umd',
    },
    umdNamedDefine: true,
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        include: path.resolve(__dirname, 'lib', 'src'),
        loader: 'babel-loader',
        exclude: /(node_modules)/,
      },
      {
        test: /\.css$/,
        use: ['style-loader', 'css-loader'],
      },

      {
        test: /\.(png|svg|jpg|gif)$/,
        type: 'asset',
        parser: {
          dataUrlCondition: {
            maxSize: 8192,
          },
        },
      },
    ],
  },
  resolve: {
    extensions: ['*', '.js'],
    cacheWithContext: false,
  },
  plugins: [
    new webpack.EnvironmentPlugin({
      NODE_ENV: 'production',
    }),
    new MiniCssExtractPlugin({
      filename: 'style.css',
    }),
  ],
};
