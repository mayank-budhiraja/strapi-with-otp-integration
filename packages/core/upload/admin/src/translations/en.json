{
  "bulk.select.label": "Select all assets",
  "button.next": "Next",
  "checkControl.crop-duplicate": "Duplicate & crop the asset",
  "checkControl.crop-original": "Crop the original asset",
  "control-card.add": "Add",
  "control-card.cancel": "Cancel",
  "control-card.copy-link": "Copy link",
  "control-card.crop": "Crop",
  "control-card.download": "Download",
  "control-card.edit": "Edit",
  "control-card.replace-media": "Replace Media",
  "control-card.save": "Save",
  "control-card.stop-crop": "Stop cropping",
  "filter.add": "Add filter",
  "form.button.replace-media": "Replace media",
  "form.input.decription.file-alt": "This text will be displayed if the asset canâ€™t be shown.",
  "form.input.label.file-alt": "Alternative text",
  "form.input.label.file-caption": "Caption",
  "form.input.label.file-name": "File name",
  "form.upload-url.error.url.invalid": "One URL is invalid",
  "form.upload-url.error.url.invalids": "{number} URLs are invalids",
  "header.actions.add-assets": "Add new assets",
  "header.actions.upload-assets": "Upload assets",
  "header.actions.upload-new-asset": "Upload new asset",
  "header.content.assets-empty": "No assets",
  "header.content.assets-multiple": "{number} assets",
  "header.content.assets-single": "1 asset",
  "input.button.label": "Browse files",
  "input.label": "Drag & Drop here or",
  "input.label-bold": "Drag & drop",
  "input.label-normal": "to upload or",
  "input.placeholder": "Click to add an asset or drag & drop a file in this area",
  "input.placeholder.icon": "Drop the asset in this zone",
  "input.url.description": "Separate your URL links by a carriage return.",
  "input.url.label": "URL",
  "list.asset.at.finished": "The assets have finished loading.",
  "list.asset.load": "Loading the asset list.",
  "list.assets-empty.search": "No result found",
  "list.assets-empty.subtitle": "Add one to the list.",
  "list.assets-empty.title": "There are no assets yet",
  "list.assets-empty.title-withSearch": "There are no assets with the applied filters",
  "list.assets.empty": "Upload your first assets...",
  "list.assets.empty.no-permissions": "The asset list is empty.",
  "list.assets.loading-asset": "Loading the preview for the media: {path}",
  "list.assets.not-supported-content": "No preview available",
  "list.assets.preview-asset": "Preview for the video at path {path}",
  "list.assets.selected": "{number, plural, =0 {No asset} one {1 asset} other {# assets}} ready to upload",
  "list.assets.type-not-allowed": "This type of file is not allowed.",
  "mediaLibraryInput.actions.nextSlide": "Next slide",
  "mediaLibraryInput.actions.previousSlide": "Previous slide",
  "mediaLibraryInput.placeholder": "Click to add an asset or drag and drop one in this area",
  "mediaLibraryInput.slideCount": "{n} of {m} slides",
  "modal.file-details.date": "Date",
  "modal.file-details.dimensions": "Dimensions",
  "modal.file-details.extension": "Extension",
  "modal.file-details.size": "Size",
  "modal.header.browse": "Upload assets",
  "modal.header.file-detail": "Details",
  "modal.header.pending-assets": "Pending assets",
  "modal.header.select-files": "Selected files",
  "modal.nav.browse": "browse",
  "modal.nav.computer": "From computer",
  "modal.nav.selected": "selected",
  "modal.nav.url": "From url",
  "modal.remove.success-label": "The asset has been successfully removed.",
  "modal.selected-list.sub-header-subtitle": "Drag & drop to reorder the assets in the field",
  "modal.upload-list.footer.button": "Upload {number, plural, one {# asset} other {# assets}} to the library",
  "modal.upload-list.sub-header-subtitle": "Manage the assets before adding them to the Media Library",
  "modal.upload-list.sub-header.button": "Add more assets",
  "modal.upload.cancelled": "Upload manually aborted.",
  "page.title": "Settings - Media Library",
  "permissions.not-allowed.update": "You are not allowed to edit this file.",
  "plugin.description.long": "Media file management.",
  "plugin.description.short": "Media file management.",
  "plugin.name": "Media Library",
  "search.clear.label": "Clear the search",
  "search.label": "Search for an asset",
  "search.placeholder": "e.g: the first dog on the moon",
  "settings.blockTitle": "Asset Management",
  "settings.form.autoOrientation.description": "Enabling this option will automatically rotate the image according to EXIF orientation tag.",
  "settings.form.autoOrientation.label": "Auto orientation",
  "settings.form.responsiveDimensions.description": "Enabling this option will generate multiple formats (small, medium and large) of the uploaded asset.",
  "settings.form.responsiveDimensions.label": "Responsive friendly upload",
  "settings.form.sizeOptimization.description": "Enabling this option will reduce the image size and slightly reduce its quality.",
  "settings.form.sizeOptimization.label": "Size optimization",
  "settings.form.videoPreview.description": "It will generate a six-second preview of the video (GIF)",
  "settings.form.videoPreview.label": "Preview",
  "settings.header.label": "Media Library",
  "settings.section.doc.label": "Doc",
  "settings.section.image.label": "Image",
  "settings.section.video.label": "Video",
  "settings.sub-header.label": "Configure the settings for the Media Library",
  "sort.created_at_asc": "Oldest uploads",
  "sort.created_at_desc": "Most recent uploads",
  "sort.label": "Sort by",
  "sort.name_asc": "Alphabetical order (A to Z)",
  "sort.name_desc": "Reverse alphabetical order (Z to A)",
  "sort.updated_at_asc": "Oldest updates",
  "sort.updated_at_desc": "Most recent updates",
  "tabs.title": "How do you want to upload your assets?",
  "window.confirm.close-modal.file": "Are you sure? Your changes will be lost.",
  "window.confirm.close-modal.files": "Are you sure? You have some files that have not been uploaded yet."
}
