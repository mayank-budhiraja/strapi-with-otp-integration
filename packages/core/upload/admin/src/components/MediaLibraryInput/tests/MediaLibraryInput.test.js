import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { QueryClient, QueryClientProvider } from 'react-query';
import { NotificationsProvider } from '@strapi/helper-plugin';
import { MediaLibraryInput } from '..';
import en from '../../../translations/en.json';

jest.mock('../../../utils/downloadFile');

jest.mock('../../../utils', () => ({
  ...jest.requireActual('../../../utils'),
  getTrad: x => x,
}));

jest.mock('react-intl', () => ({
  FormattedMessage: ({ id }) => id,
  useIntl: () => ({ formatMessage: jest.fn(({ id }) => en[id] || id) }),
}));
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
    },
  },
});

const renderCompo = (props = { intlLabel: { id: 'default', defaultMessage: 'default message' } }) =>
  render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <NotificationsProvider toggleNotification={jest.fn()}>
          <MediaLibraryInput {...props} />
        </NotificationsProvider>
      </ThemeProvider>
    </QueryClientProvider>,
    { container: document.body }
  );

describe('<MediaLibraryInput />', () => {
  it('renders and matches the snapshot', () => {
    const { container } = renderCompo();

    expect(container).toMatchSnapshot();
  });
});
