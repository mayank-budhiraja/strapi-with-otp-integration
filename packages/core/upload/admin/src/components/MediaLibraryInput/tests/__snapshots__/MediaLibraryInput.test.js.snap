// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaLibraryInput /> renders and matches the snapshot 1`] = `
.c18 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c12 {
  width: 100%;
  height: 100%;
}

.c13 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c14 {
  color: #4945ff;
  margin-bottom: 12px;
  width: 30px;
  height: 24px;
}

.c15 path {
  fill: #4945ff;
}

.c16 {
  font-weight: 600;
  color: #666687;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c4 {
  background: #f6f6f9;
  padding: 8px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
}

.c5 {
  position: relative;
}

.c7 {
  padding-right: 8px;
  padding-left: 8px;
  width: 100%;
}

.c9 {
  height: 124px;
}

.c2 {
  font-weight: 600;
  color: #32324d;
  font-size: 0.75rem;
  line-height: 1.33;
}

.c0 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  display: grid;
  grid-template-columns: auto 1fr auto;
  grid-template-areas: 'startAction slides endAction';
}

.c8 {
  grid-area: slides;
}

.c1 > * {
  margin-top: 0;
  margin-bottom: 0;
}

.c1 > * + * {
  margin-top: 4px;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c17 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<body>
  <div>
    <div
      class="c0 c1"
      spacing="1"
    >
      <label
        class="c2"
        for="carouselinput-1"
      >
        <div
          class="c3"
        >
          default
        </div>
      </label>
      <div
        class=""
        id="carouselinput-1"
      >
        <div
          class="c4"
        >
          <section
            aria-label="default"
            aria-roledescription="carousel"
            class="c5 c6"
          >
            <div
              aria-live="polite"
              class="c7 c8"
              width="100%"
            >
              <div
                aria-label="upload.mediaLibraryInput.slideCount"
                aria-roledescription="slide"
                class="c9 c10 c11"
                height="124px"
                role="group"
              >
                <button
                  class="c12 c13"
                  height="100%"
                  style="cursor: pointer;"
                  type="button"
                  width="100%"
                >
                  <svg
                    aria-hidden="true"
                    class="c14 c15"
                    fill="none"
                    height="24px"
                    viewBox="0 0 24 20"
                    width="30px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21.569 2.398H7.829v1.586h13.74c.47 0 .826.5.826 1.094v9.853l-2.791-3.17a2.13 2.13 0 00-.74-.55 2.214 2.214 0 00-.912-.196 2.215 2.215 0 00-.912.191 2.131 2.131 0 00-.74.546l-2.93 3.385-2.973-3.36a2.147 2.147 0 00-.741-.545 2.228 2.228 0 00-1.824.007c-.286.13-.538.319-.739.553l-2.931 3.432V7.653H2.51v9.894c.023.153.06.304.108.452v.127l.041.095c.057.142.126.28.207.412l.099.15c.074.107.157.207.247.302l.124.119c.13.118.275.222.43.309h.024c.36.214.775.327 1.198.325h16.515c.36-.004.716-.085 1.039-.24.323-.153.606-.375.827-.648a2.78 2.78 0 00.504-.888c.066-.217.108-.44.124-.666V5.078a2.497 2.497 0 00-.652-1.81 2.706 2.706 0 00-1.776-.87z"
                      fill="#32324D"
                    />
                    <path
                      d="M12.552 9.199c.912 0 1.651-.71 1.651-1.585 0-.876-.74-1.586-1.651-1.586-.912 0-1.652.71-1.652 1.586 0 .875.74 1.585 1.652 1.585zM3.303 6.408h.826V3.997h2.477v-.793-.793H4.129V0h-.826c-.219 0-.85.002-.826 0v2.411H0v1.586h2.477v2.41h.826z"
                      fill="#32324D"
                    />
                  </svg>
                  <span
                    class="c16 c17"
                    style="text-align: center;"
                  >
                    upload.mediaLibraryInput.placeholder
                  </span>
                </button>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c18"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</body>
`;
